### PR TITLE
Fix entrypoint

### PR DESCRIPTION
**Pull request by nyan-ko**
Reopened PR. Correctly fixes initial game data request from game window, and swapping to the game window

### CODE REVIEW

**Review by lilyxmeng — DISMISSED** [review]

File: src/main/java/ui/windows/game/GameController.java [hunk 1]
```diff
@@ -109,4 +115,11 @@ private GameDisplayData getGameDisplayData(PlayerGameResponseModel response) {
 
         return new GameDisplayData(currentPlayer, cards, new CardDisplayData(response.getCurrentCard().getSuit(), response.getCurrentCard().getRank()));
     }
+
+    /**
+     * Helper method to update the current owner of the game.
```
Comment (meiiph) [inline src/main/java/ui/windows/game/GameController.java:120]: I would change the wording of this, owner is a bit ambiguous.

File: src/main/java/ui/windows/game/GamePlayDelegator.java [hunk 2]
```diff
@@ -17,21 +17,12 @@ public class GamePlayDelegator implements ActionListener {
...
      * Fires when the user requests to play a card.
      */
     @Override
     public void actionPerformed(ActionEvent e) {
-        JButton button = (JButton) e.getSource();
-        JPanel panel = (JPanel) button.getParent().getParent().getParent().getParent();
-        GameDisplay display = (GameDisplay) panel;
-        String currentPlayer = display.getCurrentPlayer();
-        controller.setSelectedOwner(currentPlayer);
         controller.playCard();
```
Comment (meiiph) [inline src/main/java/ui/windows/game/GamePlayDelegator.java:26]: So much cleaner this way! 👍

File: src/main/java/use_cases/GameState.java [hunk 3]
```diff
@@ -98,4 +97,14 @@ public boolean getHasWinner() {
     public HashMap<Player, Integer> getPlayersAndCards() {
         return playersAndCards;
     }
+
+    private HashMap<Player, Integer> updatePlayersAndCards(ArrayList<Player> players) {
```
Comment (meiiph) [inline src/main/java/use_cases/GameState.java:101]: Documentation please. :)

File: src/main/java/use_cases/GameState.java [hunk 4]
```diff
@@ -98,4 +97,14 @@ public boolean getHasWinner() {
...
+        HashMap<Player, Integer> playersAndCards = new HashMap<>();
```
Comment (meiiph) [inline src/main/java/use_cases/GameState.java:102]: This is already initialized at the top, see line 19, is it necessary to create a new HashMap again?

Comment (nyan-ko) [reply src/main/java/use_cases/GameState.java:102]: its returning a local variable that happens to have the same name, ill change it so its more clear

**Review by meiiph — APPROVED** [review]
Looks good, you did some documentation before I could comment on it. :) The new code looks great, really cleans up some lengthier chunks of code from before.

**Review by lilyxmeng — APPROVED** [review]
The code looks much cleaner and organized, approved!

**Review by SMateen21 — APPROVED** [review]
Great job on making the code much cleaner.

**Pull request merged**